client/blocks/stats-navigation/index.js

```diff
@@ -234,50 +238,84 @@ class StatsNavigation extends Component {
 		return (
 			<div className={ wrapperClass }>
 				{ siteId && <QueryJetpackModules siteId={ siteId } /> }
-				<SectionNav selectedText={ label }>
-					<NavTabs selectedText={ label }>
-						{ Object.keys( navItems )
+				{ isNavigationImprovementEnabled ? (
+					<TabPanel
+						className="stats-navigation__tabs"
+						tabs={ Object.keys( navItems )
 							.filter( this.isValidItem )
 							.map( ( item ) => {
 								const navItem = navItems[ item ];
 								const intervalPath = navItem.showIntervals ? `/${ interval || 'day' }` : '';
 								const itemPath = `${ navItem.path }${ intervalPath }${ slugPath }`;
-								const className = 'stats-navigation__' + item;
-								if ( item === 'store' && config.isEnabled( 'is_running_in_jetpack_site' ) ) {
-									return (
-										<NavItem
-											className={ className }
-											key={ item }
-											onClick={ () =>
-												( window.location.href = `${ this.props.adminUrl }admin.php?page=wc-admin&path=%2Fanalytics%2Foverview` )
-											}
-											selected={ false }
-										>
-											{ navItem.label }
-										</NavItem>
-									);
-								}
-								return (
-									<NavItem
-										className={ className }
-										key={ item }
-										path={ itemPath }
-										selected={ selectedItem === item }
-									>
-										{ navItem.label }
-										{ navItem.paywall && showLock && ' 🔒' }
-									</NavItem>
-								);
-							} ) }
-					</NavTabs>
 
-					{ isLegacy && showIntervals && (
-						<Intervals selected={ interval } pathTemplate={ pathTemplate } />
-					) }
-				</SectionNav>
-
-				{ isLegacy && showIntervals && (
-					<Intervals selected={ interval } pathTemplate={ pathTemplate } standalone />
+								return {
+									name: item,
+									title: navItem.label + ( navItem.paywall && showLock ? ' 🔒' : '' ),
+									className: 'stats-navigation__' + item,
+									path: itemPath,
+								};
+							} ) }
+						onSelect={ ( tab ) => {
+							if ( tab.name === 'store' && config.isEnabled( 'is_running_in_jetpack_site' ) ) {
+								window.location.href = `${ this.props.adminUrl }admin.php?page=wc-admin&path=%2Fanalytics%2Foverview`;
+								return;
+							}
+							if ( tab.path ) {
```

Comment:
When I select tabs, it doesn't do anything atm, is it expect?

Also we might want to use `page` to direct users to pages rather than manipulating `location.href`.

Reply:
hmm that's odd. It was working okay for me but maybe it was a lucky glitch on my end. Thanks for pointing this out I will investigate

Reply:
should be working better now :)